public/app/features/dashboard-scene/serialization/transformSceneToSaveModelSchemaV2.ts

```diff
@@ -418,114 +419,135 @@ function getAnnotations(state: DashboardSceneState, dsReferencesMapping?: DSRefe
     if (!(layer instanceof dataLayers.AnnotationsDataLayer)) {
       continue;
     }
+
+    // TODO: in what scenario would we need this?
     const datasource = getElementDatasource(layer, layer.state.query, 'annotation', undefined, dsReferencesMapping);
```

Comment:
@axelavargas - one more question - why is this being resolved from the mapping here?